File: tiny-money-backend/src/main/java/com/mwpro/tinymoney/models/dtos/AddTransactionDto.java
package com.mwpro.tinymoney.models.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.Set;

public class AddTransactionDto {
    @NotNull
    private LocalDate transactionDate;
    @NotNull
    private BigDecimal amount;
    @NotNull
    private Integer subcategoryId;
    private Set<TagDto> tags;

    @JsonProperty(value="isExpense")
    private Boolean isExpense;

    @NotNull
    private VendorDto vendor;

    private String description;

    public LocalDate getTransactionDate() {
        return transactionDate;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public Integer getSubcategoryId() {
        return subcategoryId;
    }

    public Boolean getIsExpense() {
        return isExpense;
    }

    public Set<TagDto> getTags() {
        return tags;
    }

    public VendorDto getVendor() {
        return vendor;
    }

    public void setVendor(VendorDto vendor) {
        this.vendor = vendor;
    }

    public String getDescription() {
        return description;
    }
}

File: backend/MW.TinyMoney.Api/Transaction/ApiModels/AddTransactionDTO.cs
using System;
using System.Collections.Generic;
using MW.TinyMoney.Api.Buffer.ApiModels;

namespace MW.TinyMoney.Api.Transaction.ApiModels
{
    public class AddTransactionDto
    {
        public AddTransactionDto()
        {
            Tags = new List<TagDto>();
        }

        public decimal Amount { get; set; }
        public bool IsExpense { get; set; }
        public DateTime TransactionDate { get; set; }
        public string Description { get; set; }

        public VendorDto Vendor { get; set; }
        public int SubcategoryId { get; set; }
        public IEnumerable<TagDto> Tags { get; set; }
    }
}